streams-tests/shared/src/test/scala/zio/stream/experimental/ZSinkSpec.scala

```diff
@@ -770,6 +772,101 @@ object ZSinkSpec extends ZIOBaseSpec {
                 }
             }
           }
+        ),
+        suite("iso_8859_1")(
+          test("ISO-8859-1 strings")(checkM(Gen.iso_8859_1) { s =>
+            ZStream
+              .fromChunk(Chunk.fromArray(s.getBytes(StandardCharsets.ISO_8859_1)))
+              .transduce(ZSink.iso_8859_1Decode.map(_.getOrElse("")))
+              .mkString
+              .map(result => assertTrue(result == s))
+          })
+        ),
+        suite("utf16BEDecode")(
+          test("regular strings") {
+            checkM(Gen.const("").map(s => Chunk.fromArray(s.getBytes(StandardCharsets.UTF_16BE))).map(Chunk.single)) {
+              byteChunks =>
+                ZStream
+                  .fromChunks(byteChunks.toList: _*)
+                  .transduce(ZSink.utf16BEDecode.map(_.getOrElse("")))
+                  .mkString
+                  .map { result =>
+                    assertTrue(byteChunks.flatten == Chunk.fromArray(result.getBytes(StandardCharsets.UTF_16BE)))
+                  }
+            }
+          }
+        ),
+        suite("utf16FEDecode")(
+          test("regular strings") {
+            checkM(Gen.anyString.map(s => Chunk.fromArray(s.getBytes(StandardCharsets.UTF_16LE))).map(Chunk.single)) {
+              byteChunks =>
+                ZStream
+                  .fromChunks(byteChunks.toList: _*)
```

Comment:
Instead of wrapping the `Chunk` in another `Chunk` and using `fromChunks` you could just use `ZIO.fromChunk` on the chunk you get from the array.